In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('../data/oral_cancer_prediction_dataset.csv')
print("Dataset Loaded Successfully")

Dataset Loaded Successfully


In [3]:
print(f'Shape of Dataset: {df.shape}')

Shape of Dataset: (84922, 25)


Shape of Dataset: (84922, 25)

In [4]:
print(df.columns.to_list())

['ID', 'Country', 'Age', 'Gender', 'Tobacco Use', 'Alcohol Consumption', 'HPV Infection', 'Betel Quid Use', 'Chronic Sun Exposure', 'Poor Oral Hygiene', 'Diet (Fruits & Vegetables Intake)', 'Family History of Cancer', 'Compromised Immune System', 'Oral Lesions', 'Unexplained Bleeding', 'Difficulty Swallowing', 'White or Red Patches in Mouth', 'Tumor Size (cm)', 'Cancer Stage', 'Treatment Type', 'Survival Rate (5-Year, %)', 'Cost of Treatment (USD)', 'Economic Burden (Lost Workdays per Year)', 'Early Diagnosis', 'Oral Cancer (Diagnosis)']


['ID', 'Country', 'Age', 'Gender', 'Tobacco Use', 'Alcohol Consumption', 'HPV Infection', 'Betel Quid Use', 'Chronic Sun Exposure', 'Poor Oral Hygiene', 'Diet (Fruits & Vegetables Intake)', 'Family History of Cancer', 'Compromised Immune System', 'Oral Lesions', 'Unexplained Bleeding', 'Difficulty Swallowing', 'White or Red Patches in Mouth', 'Tumor Size (cm)', 'Cancer Stage', 'Treatment Type', 'Survival Rate (5-Year, %)', 'Cost of Treatment (USD)', 'Economic Burden (Lost Workdays per Year)', 'Early Diagnosis', 'Oral Cancer (Diagnosis)']


In [5]:
df.head()

,ID,Country,Age,Gender,Tobacco Use,Alcohol Consumption,HPV Infection,Betel Quid Use,Chronic Sun Exposure,Poor Oral Hygiene,...,Difficulty Swallowing,White or Red Patches in Mouth,Tumor Size (cm),Cancer Stage,Treatment Type,"Survival Rate (5-Year, %)",Cost of Treatment (USD),Economic Burden (Lost Workdays per Year),Early Diagnosis,Oral Cancer (Diagnosis)
0,1,Italy,36,Female,Yes,Yes,Yes,No,No,Yes,...,No,No,0.000000,0,No Treatment,100.000000,0.00,0,No,No
1,2,Japan,64,Male,Yes,Yes,Yes,No,Yes,Yes,...,No,No,1.782186,1,No Treatment,83.340103,77772.50,177,No,Yes
2,3,UK,37,Female,No,Yes,No,No,Yes,Yes,...,No,Yes,3.523895,2,Surgery,63.222871,101164.50,130,Yes,Yes
3,4,Sri Lanka,55,Male,Yes,Yes,No,Yes,No,Yes,...,No,No,0.000000,0,No Treatment,100.000000,0.00,0,Yes,No
4,5,South Africa,68,Male,No,No,No,No,No,Yes,...,No,No,2.834789,3,No Treatment,44.293199,45354.75,52,No,Yes


In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 84922 entries, 0 to 84921
Data columns (total 25 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   ID                                        84922 non-null  int64  
 1   Country                                   84922 non-null  str    
 2   Age                                       84922 non-null  int64  
 3   Gender                                    84922 non-null  str    
 4   Tobacco Use                               84922 non-null  str    
 5   Alcohol Consumption                       84922 non-null  str    
 6   HPV Infection                             84922 non-null  str    
 7   Betel Quid Use                            84922 non-null  str    
 8   Chronic Sun Exposure                      84922 non-null  str    
 9   Poor Oral Hygiene                         84922 non-null  str    
 10  Diet (Fruits & Vegetables Intake)         849

<class 'pandas.DataFrame'>
RangeIndex: 84922 entries, 0 to 84921
Data columns (total 25 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   ID                                        84922 non-null  int64  
 1   Country                                   84922 non-null  str    
 2   Age                                       84922 non-null  int64  
 3   Gender                                    84922 non-null  str    
 4   Tobacco Use                               84922 non-null  str    
 5   Alcohol Consumption                       84922 non-null  str    
 6   HPV Infection                             84922 non-null  str    
 7   Betel Quid Use                            84922 non-null  str    
 8   Chronic Sun Exposure                      84922 non-null  str    
 9   Poor Oral Hygiene                         84922 non-null  str    
 10  Diet (Fruits & Vegetables Intake)         84922 non-null  str    
 11  Family History of Cancer                  84922 non-null  str    
 12  Compromised Immune System                 84922 non-null  str    
 13  Oral Lesions                              84922 non-null  str    
 14  Unexplained Bleeding                      84922 non-null  str    
 15  Difficulty Swallowing                     84922 non-null  str    
 16  White or Red Patches in Mouth             84922 non-null  str    
 17  Tumor Size (cm)                           84922 non-null  float64
 18  Cancer Stage                              84922 non-null  int64  
 19  Treatment Type                            84922 non-null  str    
...
 23  Early Diagnosis                           84922 non-null  str    
 24  Oral Cancer (Diagnosis)                   84922 non-null  str    
dtypes: float64(3), int64(4), str(18)
memory usage: 16.2 MB
Output is truncated. View as a scrollable element or open in a text editor. Adjust cell output settings...

In [7]:
df.describe()

,ID,Age,Tumor Size (cm),Cancer Stage,"Survival Rate (5-Year, %)",Cost of Treatment (USD),Economic Burden (Lost Workdays per Year)
count,84922.000000,84922.000000,84922.000000,84922.000000,84922.000000,84922.000000,84922.000000
mean,42461.500000,54.509444,1.747294,1.118756,79.503364,39109.881244,52.028391
std,24515.014117,10.014839,2.028199,1.340418,26.483746,44710.687054,60.389796
min,1.000000,15.000000,0.000000,0.000000,10.000750,0.000000,0.000000
25%,21231.250000,48.000000,0.000000,0.000000,65.233425,0.000000,0.000000
50%,42461.500000,55.000000,0.000000,0.000000,100.000000,0.000000,0.000000
75%,63691.750000,61.000000,3.480075,2.000000,100.000000,76468.437500,104.000000
max,84922.000000,101.000000,5.999899,4.000000,100.000000,159988.000000,179.000000


In [9]:
df.isnull().sum()

ID                                          0
Country                                     0
Age                                         0
Gender                                      0
Tobacco Use                                 0
Alcohol Consumption                         0
HPV Infection                               0
Betel Quid Use                              0
Chronic Sun Exposure                        0
Poor Oral Hygiene                           0
Diet (Fruits & Vegetables Intake)           0
Family History of Cancer                    0
Compromised Immune System                   0
Oral Lesions                                0
Unexplained Bleeding                        0
Difficulty Swallowing                       0
White or Red Patches in Mouth               0
Tumor Size (cm)                             0
Cancer Stage                                0
Treatment Type                              0
Survival Rate (5-Year, %)                   0
Cost of Treatment (USD)           

In [10]:
df.duplicated().sum()

np.int64(0)

In [11]:
df['Oral Cancer (Diagnosis)'].value_counts()

Oral Cancer (Diagnosis)
No     42573
Yes    42349
Name: count, dtype: int64

In [12]:
df['Oral Cancer (Diagnosis)'].value_counts(normalize=True)*100

Oral Cancer (Diagnosis)
No     50.131886
Yes    49.868114
Name: proportion, dtype: float64

In [13]:
df.dtypes

ID                                            int64
Country                                         str
Age                                           int64
Gender                                          str
Tobacco Use                                     str
Alcohol Consumption                             str
HPV Infection                                   str
Betel Quid Use                                  str
Chronic Sun Exposure                            str
Poor Oral Hygiene                               str
Diet (Fruits & Vegetables Intake)               str
Family History of Cancer                        str
Compromised Immune System                       str
Oral Lesions                                    str
Unexplained Bleeding                            str
Difficulty Swallowing                           str
White or Red Patches in Mouth                   str
Tumor Size (cm)                             float64
Cancer Stage                                  int64
Treatment Ty

In [14]:
categorical_cols = df.select_dtypes(include='object').columns.tolist()
for col in categorical_cols:
    print(f"\n{col}: {df[col].unique()}")


Country: <StringArray>
[       'Italy',        'Japan',           'UK',    'Sri Lanka',
 'South Africa',       'Taiwan',          'USA',      'Germany',
       'France',    'Australia',       'Brazil',     'Pakistan',
        'Kenya',       'Russia',      'Nigeria',        'Egypt',
        'India']
Length: 17, dtype: str

Gender: <StringArray>
['Female', 'Male']
Length: 2, dtype: str

Tobacco Use: <StringArray>
['Yes', 'No']
Length: 2, dtype: str

Alcohol Consumption: <StringArray>
['Yes', 'No']
Length: 2, dtype: str

HPV Infection: <StringArray>
['Yes', 'No']
Length: 2, dtype: str

Betel Quid Use: <StringArray>
['No', 'Yes']
Length: 2, dtype: str

Chronic Sun Exposure: <StringArray>
['No', 'Yes']
Length: 2, dtype: str

Poor Oral Hygiene: <StringArray>
['Yes', 'No']
Length: 2, dtype: str

Diet (Fruits & Vegetables Intake): <StringArray>
['Low', 'High', 'Moderate']
Length: 3, dtype: str

Family History of Cancer: <StringArray>
['No', 'Yes']
Length: 2, dtype: str

Compromised Immune Sys

C:\Users\Athi\AppData\Local\Temp\ipykernel_31768\2809860531.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_cols = df.select_dtypes(include='object').columns.tolist()


In [15]:
# Columns to drop
drop_cols = [
    'ID',
    'Tumor Size (cm)',
    'Cancer Stage',
    'Treatment Type',
    'Survival Rate (5-Year, %)',
    'Cost of Treatment (USD)',
    'Economic Burden (Lost Workdays per Year)',
    'Early Diagnosis'
]

# Check remaining columns after drop
remaining = [col for col in df.columns if col not in drop_cols]
print("Remaining columns:", remaining)
print("Total remaining:", len(remaining))

Remaining columns: ['Country', 'Age', 'Gender', 'Tobacco Use', 'Alcohol Consumption', 'HPV Infection', 'Betel Quid Use', 'Chronic Sun Exposure', 'Poor Oral Hygiene', 'Diet (Fruits & Vegetables Intake)', 'Family History of Cancer', 'Compromised Immune System', 'Oral Lesions', 'Unexplained Bleeding', 'Difficulty Swallowing', 'White or Red Patches in Mouth', 'Oral Cancer (Diagnosis)']
Total remaining: 17
